server/grpc_service.go

```diff
@@ -710,6 +714,9 @@ func (s *GrpcServer) IsSnapshotRecovering(ctx context.Context, _ *pdpb.IsSnapsho
 			return nil, errs.ErrGRPCRateLimitExceeded(err)
 		}
 	}
+	if s.IsClosed() {
```

Comment:
maybe placing this check to the head of this function is better?

Reply:
Other functions' checking `IsClosed()` is inside `unaryMiddleware`. All of the `unaryMiddleware` is behind `Ratelimit` component. So here it is consistent with other places.